s3parq/fetch_parq.py

```diff
@@ -329,21 +332,26 @@ def _get_filtered_key_list(typed_parts: dict, filters, key) -> List[str]:
     are set ie all non-matching partitions are excluded.
     '''
     filter_keys = []
+    matched_parts = OrderedDict()
+    matched_parts.keys = typed_parts.keys()
+    matched_part_vals = set()
 
     for part, part_values in typed_parts.items():
-        for f in filters:
-            if (f['partition'] == part):
-                comparison = OPS[f['comparison']]
-                for v in f['values']:
-                    typed_parts[part] = set(filter(
-                        lambda x: comparison(x, v),
-                        typed_parts[part]
-                    )
-                    )
-
-    def construct_paths(typed_parts, previous_fil_keys: List[str]) -> None:
-        if len(typed_parts) > 0:
-            part = typed_parts.popitem(last=False)
+        matched_part_vals.clear()
+        fil = next((f for f in filters if f['partition'] == part), False)
+        if fil:
+            comparison = OPS[fil['comparison']]
+            for v in fil['values']:
+                for x in part_values:
+                    if comparison(x, v):
+                        matched_part_vals.add(x)
+            matched_parts[part] = matched_part_vals.copy()
+        else:
+            matched_parts[part] = part_values
+            
+    def construct_paths(matched_parts, previous_fil_keys: List[str]) -> None:
```

Comment:
What `type` is `matched_parts` supposed to be?  if we know `previous_fil_keys` is expected to be a `List[str]`, we should try to be consistent and be explicit about what type this variable will be

Reply:
matched_parts is defined in `_get_filtered_key_list()` to be an OrderedDict, similar to typed_parts (at the top of the function). While typed_parts contains _all_ the partition values for a particular dataset, matched_parts removes partition values based on the filters passed to the function.